### PR TITLE
feat(*): update shared folder with latest changes

### PR DESCRIPTION
**Pull request by fulopdaniel**
The last few weeks we were still adding new code to guideline-blocks/shared.

After this PR is merged this package will be the single source of shared components/logic within the guideline blocks.

### CODE REVIEW

File: packages/guideline-blocks-settings/src/components/RichTextEditor/RichTextEditor.tsx [hunk 1]
```diff
@@ -14,17 +15,20 @@ export const RichTextEditor = ({
...
+        onTextChange && newContent !== value && onTextChange(newContent);
+        setShouldPreventPageLeave(false);
+    };
+
     useEffect(() => {
-        const unloadHandler = (event: BeforeUnloadEvent) => {
-            event.preventDefault();
-            return (event.returnValue = 'Unprocessed changes');
+        const unloadHandler = (e: BeforeUnloadEvent) => {
+            e.preventDefault();
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/RichTextEditor/RichTextEditor.tsx:30]: revert

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ export const QuoteMarkupElementNode = ({ element, attributes, children, styles }
     return (
         <blockquote
             {...attributes}
-            className={merge([align && alignmentClassnames[align], getColumnBreakClasses(element), 'quote'])}
+            className={merge([align && alignmentClassnames[align], getColumnBreakClasses(element), 'a-quote'])}
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx:45]: why `a-quote`?

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/shared/LinkSelector/LinkSelector.tsx [hunk 3]
```diff
@@ -49,7 +49,7 @@ export const LinkSelector = ({ appBridge, url, onUrlChange }: LinkSelectorProps)
             >
                 Internal link
             </Button>
-            <Modal zIndex={200} onClose={() => closeLinkTree()} isOpen={isLinkTreeOpen} isDismissable>
+            <Modal zIndex={200000} onClose={() => closeLinkTree()} isOpen={isLinkTreeOpen} isDismissable>
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/shared/LinkSelector/LinkSelector.tsx:52]: Smelly code

File: packages/guideline-blocks-settings/src/settings/securityDownloadable.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        id: options?.id ? options.id : 'downloadable',
+        type: 'switch',
+        defaultValue: false,
+        label: 'Downloadable',
+        show: (bundle) => bundle.getBlock(securityId)?.value === Security.Custom,
+    };
+};
+
+export const getSecurityGlobalControlId = (id?: string): string => {
+    return id ? id : 'security';
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/settings/securityDownloadable.ts:32]:
```suggestion
    return id || 'security';
```

File: packages/guideline-blocks-settings/src/settings/types.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import { Color } from '@frontify/sidebar-settings';
+import { Color } from '..';
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/settings/types.ts:3]: Why changing?

File: packages/guideline-blocks-settings/src/utilities/color/isDark.ts [hunk 6]
```diff
@@ -8,6 +8,7 @@ import { Color } from '@frontify/sidebar-settings';
  * Calculates if text should be in a light color depending on color (e.g. background-color)
  *
  * @param {Object} value Object of RGBA values
+ * @param {Number} threshold Threshold for the brightness
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/utilities/color/isDark.ts:11]:
```suggestion
 * @param {Number|undefined} threshold Threshold for the brightness
```

**Review by SamuelAlev — CHANGES_REQUESTED** [review]

File: packages/guideline-blocks-settings/src/components/Attachments/AttachmentItem.tsx [hunk 7]
```diff
@@ -122,13 +123,14 @@ export const AttachmentItem = forwardRef<HTMLButtonElement, AttachmentItemProps>
                     <div
                         data-test-id="attachments-actionbar"
                         className={joinClassNames([
-                            'tw-flex tw-gap-0.5 group-focus:tw-opacity-100 focus-visible:tw-opacity-100 focus-within:tw-opacity-100 group-hover:tw-opacity-100',
+                            'tw-flex tw-gap-0.5 group-focus:tw-opacity-100 [&:has(:focus-visible)]:tw-opacity-100  group-hover:tw-opacity-100',
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/Attachments/AttachmentItem.tsx]: Thanks, fixed.

File: packages/guideline-blocks-settings/src/components/Attachments/Attachments.tsx [hunk 8]
```diff
@@ -131,90 +131,93 @@ export const Attachments = ({
...
-                                    onDragStart={handleDragStart}
-                                    onDragEnd={handleDragEnd}
-                                    modifiers={[restrictToWindowEdges]}
-                                >
-                                    <SortableContext items={internalItems} strategy={rectSortingStrategy}>
-                                        <div className="tw-border-b tw-border-b-line">
-                                            {internalItems.map((item) => (
-                                                <SortableAttachmentItem
+    return (
+        (isEditing || (internalItems?.length ?? 0) > 0) && (
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/Attachments/Attachments.tsx]: Thanks, fixed.

File: packages/guideline-blocks-settings/src/components/Attachments/types.ts [hunk 9]
```diff
@@ -3,7 +3,7 @@
 import { AppBridgeBlock, Asset } from '@frontify/app-bridge';
 
 export type AttachmentsProps = {
-    items?: Asset[];
+    items: Asset[] | undefined;
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/Attachments/types.ts]: Thanks, fixed.

File: packages/guideline-blocks-settings/src/components/BlockInjectButton/BlockInjectButton.tsx [hunk 10]
```diff
@@ -79,18 +74,21 @@ export const BlockInjectButton = ({
             ref={buttonRef}
             data-test-id="block-inject-button"
             className={joinClassNames([
-                ' tw-font-body tw-relative tw-text-[14px] tw-leading-4 tw-text-text-weak tw-border tw-flex tw-items-center tw-justify-center tw-cursor-pointer tw-gap-3 tw-w-full first:tw-rounded-tl last:tw-rounded-br',
+                'tw-font-body tw-relative tw-text-[14px] tw-leading-4 tw-border tw-flex tw-items-center tw-justify-center tw-cursor-pointer tw-gap-3 tw-w-full first:tw-rounded-tl last:tw-rounded-br',
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/BlockInjectButton/BlockInjectButton.tsx]: Thanks, fixed.

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/shared/LinkSelector/LinkSelector.tsx (hunk 3 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/shared/LinkSelector/LinkSelector.tsx]: Thanks, fixed to 1001. The link selector modal has a z index of 1000, that's why we need it higher than that.

File: packages/guideline-blocks-settings/src/helpers/index.ts [hunk 11]
```diff
@@ -1,6 +1,8 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
 export * from './convertToRichTextValue';
+export * from './customCoordinatesGetterFactory';
+export * from './downloadAsset';
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/helpers/index.ts]: Fixed, thanks.

File: packages/guideline-blocks-settings/src/settings/securityDownloadable.ts (hunk 4 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/settings/securityDownloadable.ts]: Fixed, thanks.

File: packages/guideline-blocks-settings/src/settings/types.ts (hunk 5 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/settings/types.ts]: Fixed, thanks.

File: packages/guideline-blocks-settings/src/utilities/color/isDark.ts (hunk 6 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/utilities/color/isDark.ts]: Fixed, thanks.

File: packages/guideline-blocks-settings/src/components/BlockItemWrapper/BlockItemWrapper.spec.ct.tsx [hunk 12]
```diff
@@ -37,9 +37,23 @@ describe('Block Item Wrapper', () => {
...
+                    { icon: <IconMagnifier16 />, onClick: cy.stub(), tooltip: 'Test tooltip' },
+                    { icon: <IconMagnifier16 />, onClick: cy.stub(), tooltip: 'Test tooltip' },
+                ]}
+            >
+                <div data-test-id="block-item-wrapper-child" className="tw-w-8 tw-h-8 tw-bg-red-50" />
+            </BlockItemWrapper>
+        );
+        cy.get(ToolbarButtonSelector).should('have.length', 2);
+    });
     it('should render the flyout button with the right amount of menu items', () => {
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/BlockItemWrapper/BlockItemWrapper.spec.ct.tsx:57]:
```suggestion
    });

    it('should render the flyout button with the right amount of menu items', () => {
```

File: packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar.tsx [hunk 13]
```diff
@@ -120,3 +119,4 @@ export const Toolbar = ({
         </div>
     );
 };
+export default Toolbar;
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar.tsx:122]: No export default allowed

File: packages/guideline-blocks-settings/src/components/RichTextEditor/RichTextEditor.tsx [hunk 14]
```diff
@@ -14,13 +15,16 @@ export const RichTextEditor = ({
...
     onTextChange,
-    onValueChanged,
-    updateValueOnChange,
     showSerializedText,
-    shouldPreventPageLeave,
 }: RichTextEditorProps) => {
+    const [shouldPreventPageLeave, setShouldPreventPageLeave] = useState(false);
+
+    const saveText = (newContent: string) => {
+        onTextChange && newContent !== value && onTextChange(newContent);
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/RichTextEditor/RichTextEditor.tsx:24]:
```suggestion
        if (onTextChange && newContent !== value) {
            onTextChange(newContent);
        }
```

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/LinkPlugin/utils/index.ts [hunk 15]
```diff
@@ -1,3 +1,4 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
 export * from './getUrl';
+export * from './relativeUrlregex';
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/LinkPlugin/utils/index.ts:4]: Please rename the file to camel case

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx (hunk 2 above)
Comment (SamuelAlev) [reply packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx:45]: What is `a-quote` representing? from where is that class coming from?

File: packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar.tsx (hunk 13 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/BlockItemWrapper/Toolbar.tsx]: Thanks, fixed.

File: packages/guideline-blocks-settings/src/components/RichTextEditor/RichTextEditor.tsx (hunk 14 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/RichTextEditor/RichTextEditor.tsx]: Thanks, fixed.

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/LinkPlugin/utils/index.ts (hunk 15 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/LinkPlugin/utils/index.ts]: Thanks, fixed.

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx (hunk 2 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx:45]: We use it to to inherit the right font-family if "quote" has the font-family `inherit`.

It comes from app-server's `appearance.php`

Comment (SamuelAlev) [reply packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx:45]: This is really flakey as we are not aware of it being used here, (we are currently on the path of removing the file), please rely on the CSS variables only

**Review by SamuelAlev — APPROVED** [review]

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx (hunk 2 above)
Comment (SamuelAlev) [reply packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.tsx:45]: (To be tackled in follow up PR)

**Pull request merged**